packages/backend/src/services/ProjectService/ProjectService.ts

```diff
@@ -2109,36 +2110,59 @@ export class ProjectService {
         if (!explores) {
             return [];
         }
-        const allExploreSummaries = explores.map<SummaryExplore>((explore) => {
-            if (isExploreError(explore)) {
-                return {
-                    name: explore.name,
-                    label: explore.label,
-                    tags: explore.tags,
-                    groupLabel: explore.groupLabel,
-                    errors: explore.errors,
-                    databaseName:
-                        explore.baseTable &&
-                        explore.tables?.[explore.baseTable]?.database,
-                    schemaName:
-                        explore.baseTable &&
-                        explore.tables?.[explore.baseTable]?.schema,
-                    description:
-                        explore.baseTable &&
-                        explore.tables?.[explore.baseTable]?.description,
-                };
-            }
+        const userAttributes =
+            await this.userAttributesModel.getAttributeValuesForOrgMember({
+                organizationUuid,
+                userUuid: user.userUuid,
+            });
 
-            return {
-                name: explore.name,
-                label: explore.label,
-                tags: explore.tags,
-                groupLabel: explore.groupLabel,
-                databaseName: explore.tables[explore.baseTable].database,
-                schemaName: explore.tables[explore.baseTable].schema,
-                description: explore.tables[explore.baseTable].description,
-            };
-        });
+        const allExploreSummaries = explores.reduce<SummaryExplore[]>(
+            (acc, explore) => {
+                if (isExploreError(explore)) {
+                    return [
+                        ...acc,
+                        {
+                            name: explore.name,
+                            label: explore.label,
+                            tags: explore.tags,
+                            groupLabel: explore.groupLabel,
+                            errors: explore.errors,
+                            databaseName:
+                                explore.baseTable &&
+                                explore.tables?.[explore.baseTable]?.database,
+                            schemaName:
+                                explore.baseTable &&
+                                explore.tables?.[explore.baseTable]?.schema,
+                            description:
+                                explore.baseTable &&
+                                explore.tables?.[explore.baseTable]
+                                    ?.description,
+                        },
+                    ];
+                }
+                if (
+                    doesExploreMatchRequiredAttributes(explore, userAttributes)
```

Comment:
why don't we do this with casl? It helps us to centralise access logic

```
ability.can('view', subject('Explore', payload))
```

Reply:
Yes, we could have a `Explore` and `Field` subject. 
This means every request, with authentication, would need to fetch all the explores and user attributes to generate the permission rules.
This would slow down the app significantly unless we could cache the authenticated user instead of recreating it on every request.

Reply:
It wouldn't need to do that. The explores are the subject of the authentication request (not the definition of an ability) so we don't pre-load any explores in order to build the abilities. But yes we would need the user attributes to build the abilities.

I'm creating a separate PR to clean up all permissions: https://github.com/lightdash/lightdash/issues/8601